.github/workflows/test.yaml

```diff
@@ -67,6 +67,10 @@ jobs:
           --strict-optional \
           -p pydantic_numpy
 
+      - name: Validate type-hints with pyright
+        run: |
+          poetry run pyright pydantic_numpy
+
```

Comment:
This is defined under the mypy job, the pyright portion has to be defined under a new job.